packages/roosterjs-editor-api/lib/format/toggleBlockQuote.ts

```diff
@@ -0,0 +1,128 @@
+    }
+
+    let blockElement = contentTraverser.currentBlockElement;
+    let nodeAtCursor = getNodeAtCursor(editor);
+
+    if (getListStateAtSelection(editor, nodeAtCursor) != ListState.BlockQuote) {
+        formatter = () => {
+            let nodes: Node[];
+            let startContainer: Node;
+            let startOffset: number;
+            let isRangeCollapsed = true;
+
+            if (!range.collapsed) { // If selection not collapsed, check and try to wrap nodes in selection
+                startContainer = range.startContainer;
+                startOffset = range.startOffset;
+                isRangeCollapsed = false;
+                nodes = [];
+                while (blockElement) {
+                    // Some of the nodes in selection might already in blockquote, only add the ones not in blockquote
+                    if (!getListElementAtNode(editor, blockElement.getStartNode(), BLOCKQUOTE_TAG_NAME)) {
+                        nodes = nodes.concat(blockElement.getContentNodes());
+                    }
+                    blockElement = contentTraverser.getNextBlockElement();
+                }
+            } else if (blockElement) { // Selection is collapsed and there's content in the block, move the whole block into blockquote
+                nodes = blockElement.getContentNodes();
+                startContainer = range.startContainer;
+                startOffset = range.startOffset;
+
+                // If the only content node is <BR>, we wrap it with <DIV>, otherwise the format will break. This often happens in firefox.
+                if (nodes.length == 1 && nodes[0].nodeName == 'BR') {
+                    nodes[0] = wrap(nodes[0], '<div></div>') as HTMLDivElement;
+                    startContainer = nodes[0];
+                    startOffset = NodeBoundary.Begin;
+                }
+            } else { // Selection is collapsed and blockElment is null, we need to create an empty div. In case of IE and Edge, we insert ZWS to put cursor in the div, otherwise insert BR node
+                let div = editor.getDocument().createElement('div');
+
+                if (browserData.isEdge || browserData.isIE) {
+                    div.innerHTML = ZERO_WIDTH_SPACE;
+                } else {
+                    let brNode = editor.getDocument().createElement('br');
+                    div.appendChild(brNode);
+                }
+                editor.insertNode(div, {
+                    position: ContentPosition.SelectionStart,
+                    updateCursor: true,
+                    replaceSelection: true,
+                    insertOnNewLine: false,
+                });
+                startContainer = div.firstChild;
+                startOffset = NodeBoundary.Begin;
+                nodes = [div];
+            }
+
+            let quoteElement = wrapAll(nodes, '<blockquote></blockqupte>') as HTMLQuoteElement;
+            if (!styler) {
+                styler = (element: HTMLElement) => {
```

Comment:
Suggest to move the default styler out of this function.

let defaultStyler = (element: HTMLElement) => {
     element.style.borderLeft = "";
     ...
};

In the declaration of this function:
function toggleBlockQuote(
    editor: Editor, 
    styler: (element: HTMLElement) => void = defaultStyler
) { ... }

Reply:
Changed.